applicationset/controllers/applicationset_controller.go

```diff
@@ -938,6 +939,21 @@ func (r *ApplicationSetReconciler) buildAppDependencyList(logCtx *log.Entry, app
 	return appDependencyList, appStepMap
 }
 
+func getMatchExpressionValues(matchExpression argov1alpha1.ApplicationMatchExpression) ([]string, error) {
+	if len(matchExpression.ValuesString) > 0 && matchExpression.ValuesString != "" {
+		return nil, errors.New("only one of 'values' or 'valuesString' must be set")
+	}
+	if matchExpression.ValuesString != "" {
+		values := strings.Split(matchExpression.ValuesString, ",")
```

Comment:
It could be worth trimming the values between the ,.

Like if we have "value1,  value2, value3  , value4   "

Reply:
Resolved, I check using a regex. Specific tests related to space character between values have been implemented.